applicationset/utils/utils.go

```diff
@@ -448,6 +450,54 @@ func SanitizeName(name string) string {
 	return strings.Trim(name, "-.")
 }
 
+// SlugifyName generates a URL-friendly slug from the provided name and additional options.
+// The slug is generated in accordance with the following rules:
+// 1. The generated slug will be URL-safe and suitable for use in URLs.
+// 2. The maximum length of the slug can be specified using the `maxSize` argument.
+// 3. Smart truncation can be enabled or disabled using the `EnableSmartTruncate` argument.
+// 4. The input name can be any string value that needs to be converted into a slug.
+//
+// Args:
+// - args: A variadic number of arguments where:
+//   - The first argument (if provided) is an integer specifying the maximum length of the slug.
+//   - The second argument (if provided) is a boolean indicating whether smart truncation is enabled.
+//   - The last argument (if provided) is the input name that needs to be slugified.
+//     If no name is provided, an empty string will be used.
+//
+// Returns:
+// - string: The generated URL-friendly slug based on the input name and options.
+func SlugifyName(args ...interface{}) string {
+	// Default values for arguments
+	maxSize := 50
+	EnableSmartTruncate := true
+	name := ""
+
+	// Process the arguments
+	for idx, arg := range args {
+		switch idx {
+		case len(args) - 1:
+			name = arg.(string)
+		case 0:
+			maxSize = arg.(int)
+		case 1:
+			EnableSmartTruncate = arg.(bool)
+		default:
+			log.Errorf("Bad 'slugify' arguments.")
+		}
+	}
```

Comment:
Hm. I'm out of my element with go templating, but why the dynamic parameters?

Reply:
I wanted to give the end user the possibility to omit some arguments and give the chance to add more in the future for backward compatibility; with dynamic params all of the following is correct
- {{ cat branch | slugify }}
- {{ cat branch | slugify 50 }}
- {{ cat branch | slugify 50 false }}

Reply:
That's fancy as heck. Okay, sounds good to me! I'm not sure how error handling typically works in go template functions, but should we capture type check failures etc. and report those up as an error? Or is panicking typical for these functions?

Reply:
Here is how it throws the error, it's somehow clear that the parameters are wrong <img width="1026" alt="2023-08-24_17-39-35" src="https://github.com/argoproj/argo-cd/assets/1928890/f7b90473-c0af-4b3f-ad6a-f73d5d8a7b84">